ddtrace/internal/datadog/profiling/ddup.pyx

```diff
@@ -0,0 +1,178 @@
+            LockRelease "ProfileType::LockRelease"
+            Allocation  "ProfileType::Allocation"
+            Heap        "ProfileType::Heap"
+            All         "ProfileType::All"
+    cdef extern from "interface.hpp":
+        ctypedef signed int int64_t
+        ctypedef unsigned int uint64_t
+        void ddup_config_env(const char *env)
+        void ddup_config_service(const char *service)
+        void ddup_config_version(const char *version)
+        void ddup_config_runtime(const char *runtime)
+        void ddup_config_runtime_version(const char *runtime_version)
+        void ddup_config_profiler_version(const char *profiler_version)
+        void ddup_config_url(const char *url)
+        void ddup_config_max_nframes(int max_nframes)
+
+        void ddup_config_user_tag(const char *key, const char *val)
+        void ddup_config_sample_type(unsigned int type)
+
+        void ddup_init()
+
+        void ddup_start_sample(unsigned int nframes)
+        void ddup_push_walltime(int64_t walltime, int64_t count)
+        void ddup_push_cputime(int64_t cputime, int64_t count)
+        void ddup_push_acquire(int64_t acquire_time, int64_t count)
+        void ddup_push_release(int64_t release_time, int64_t count)
+        void ddup_push_alloc(uint64_t size, uint64_t count)
+        void ddup_push_heap(uint64_t size)
+        void ddup_push_lock_name(const char *lock_name)
+        void ddup_push_threadinfo(int64_t thread_id, int64_t thread_native_id, const char *thread_name)
+        void ddup_push_task_id(int64_t task_id)
+        void ddup_push_task_name(const char *task_name)
+        void ddup_push_span_id(int64_t span_id)
+        void ddup_push_local_root_span_id(int64_t local_root_span_id)
+        void ddup_push_trace_type(const char *trace_type)
+        void ddup_push_trace_resource_container(const char *trace_resource_container)
+        void ddup_push_exceptioninfo(const char *exception_type, int64_t count)
+        void ddup_push_class_name(const char *class_name)
+        void ddup_push_frame(const char *_name, const char *_filename, uint64_t address, int64_t line)
+        void ddup_flush_sample()
+        void ddup_set_runtime_id(const char *_id, size_t sz)
+        void ddup_upload()
+
+    def init(
+            service: str,
+            env: Optional[str],
+            version: Optional[str],
+            tags: Optional[typing.Dict[str, str]],
+            max_nframes: Optional[int],
+            url: Optional[str]) -> None:
+
+        # Default service is the main script, but what if there isn't one?
+        if not service:
+            if hasattr(__main__, "__file"):
+                service = os.path.basename(__main__.__file__)
+            else:
+                service = os.path.basename(sys.argv[0])
+        ddup_config_service(str.encode(service))
+
+        # Default env is "prod"
+        if not env:
+            env = "prod"
+        ddup_config_env(str.encode(env))
```

Comment:
Ah, thank you for calling this out.  I went back and it turns out the backend now treats `None` as a distinct type (equivalently, it populates empty fields with `None` so as to provide the same effect).

Accordingly, the correct default for the three version/env/service fields is omission.

Reply:
Fixed in 668f73d11234f60fe977b61fe7f418da10ed4afa

Reply:
For anyone following along in the comments alone
* DEFAULT_SERVICE_NAME is taken as the default service name if none is provided by the user (through environment variables or otherwise).  This behavior only relates to the setup for the libdatadog wrapper and does not interact at all with the profiler class in ddtrace
* For `service`, `env`, and `version`; for any of these parameters which are unset, the corresponding tag is omitted from the upload.  From checking in with the rest of the Profiling team, the burden is on the backend to name the service appropriately and sanitize defaults.  As such, providing defaults here (except ones that would allow the profiling data to conform with the trace data, as is the preferred defaulting behavior for `service`), would be erroneous.